operator-common/src/main/java/io/strimzi/operator/common/operator/resource/DeploymentOperator.java

```diff
@@ -105,6 +105,12 @@ protected Integer currentScale(String namespace, String name) {
         return waitFor(reconciliation, namespace, name, "observed", pollIntervalMs, timeoutMs, this::isObserved);
     }
 
+    public Future<Void> readiness(Reconciliation reconciliation, String namespace, String name, long pollIntervalMs, long timeoutMs) {
+        return getAsync(namespace, name)
+                .compose(observe -> waitForObserved(reconciliation, namespace, name, pollIntervalMs, timeoutMs)
+                .compose(ready -> super.readiness(reconciliation, namespace, name, pollIntervalMs, timeoutMs)));
```

Comment:
Rather than composing both `waitForObserved` and `readiness` we could just `waitFor(isReady && isObserved)`, thus avoiding needing a `Future`.

Reply:
I'm not sure. We want to check if it is ready after it was rolled out. I guess in theory it should work the way you suggest, but should be tested on a real deployment.

Reply:
@tombentley Hi , I didn't quite got the complete gist of this `waitFor(isReady && isObserved)`

Reply:
@ShubhamRwt look at the impl of both`waitForObserved` and `super.readiness` and you'll see they both poll the apiserver, using a method called `waitFor()`, waiting until some condition on the resource is satisfied. It should be equivalent to test for the conjunction of both conditions using a single call to `waitFor`.

Reply:
Yeah now I get it. Thanks @tombentley.